learntools/machine_learning/nbs/tut7.ipynb

```diff
@@ -0,0 +1,154 @@
+{
```

Comment:
I'm finding this last lesson hard to follow. Thoughts I had during my first readthrough:

- We're going to go through an example of submitting to a competition. Okay, *which competition*?
- Are we still working with the Melbourne housing data? Seems like the predictor names are the same as before, but the filenames are different (`train.csv` and `test.csv` instead of `melb_data.csv`)

The first 6 lessons did a great job of building on top of each other and following a coherent thread, but this one feels disconnected. 

I think part of the confusion is that the first 6 lessons were about explaining machine learning concepts, but this one is explaining the interface around a specific Kaggle feature. It almost seems like this is a piece of Kaggle documentation.

Reply:
That said, conceptually, I do like the idea of using a competition as the carrot to induce them to experiment and try to push their models further. I just wonder if this notebook should focus more on the ML stuff, with the tutorial on the competitions interface going in a separate standalone notebook (linked to from the tutorial notebook).

One example of a more ML-ish thing we could talk about here: When we submit our predictions to the competition, we see that we earned a score like 0.19734, which looks nothing like the MAEs that we were getting. Digging in, we see that this competition is ranked by Root Mean Squared Logarithmic Error. What's that? How can we measure it on our validation set? Should we re-assess our choice of parameters if we're optimizing for this new metric?